pkg/git/git.go

```diff
@@ -161,7 +161,12 @@ func GetGitPlan(
 
 		var cloneArg string
 		if parsedGitPlan.Protocol == "https" {
-			cloneArg = "https://" + parsedGitPlan.Repository
+			cloneToken := os.Getenv("GIT_CLONE_TOKEN")
+			if cloneToken == "" {
+				cloneArg = "https://" + parsedGitPlan.Repository
+			} else {
+				cloneArg = "https://" + cloneToken + ":" + parsedGitPlan.Repository
```

Comment:
consider using fmt.Sprintf for combining the strings. 

or alternatively using url package for constructing the url.